### PR TITLE
Rewrite Radar around secret types

### PR DESCRIPTION
**Pull request by Piccirello**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,16 +88,18 @@ You can instruct the CLI to output JSON by specifying the `--json` flag. This ou
...
       "fileExtension": "env"
     },
-    "keys": [
+    "secrets": [
       {
-        "key": "BpvW9qw31eXXHEGDMbERBkQ24lF6EWkUyaOgU4LG",
-        "lineNumber": 4,
-        "score": 0.9
+        "secret": "BpvW9qw31eXXHEGDMbERBkQ24lF6EWkUyaOgU4LG",
+        "type": "API Key",
```
Comment (bvallelunga) [inline README.md]: can the type be underscore and all lowercase.

File: src/filters/dictionary.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+const customDictionary = require('../customdictionary');
+const countryDictionary = require('../countrydictionary');
+const filetypes = require('../filetypes');
+const Filter = require('../objects/Filter');
+
+const name = 'Dictionary words';
+
+class CustomFilter extends Filter {
+  constructor(name) {
```
Comment (bvallelunga) [inline src/filters/dictionary.js]: One structure I saw at Uber was to have the constructor not take in the name but then set it in the super. So line 8 would not exist and instead super would be called with `"Dictionary Words"` provided.

File: README.md (hunk 1 above)
Comment (bvallelunga) [inline README.md]: can the type be underscore and all lowercase.

File: src/filters/dictionary.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    super(name);
```
Comment (bvallelunga) [inline src/filters/dictionary.js]: One structure I saw at Uber was to have the constructor not take in the name but then set it in the super. So line 8 would not exist and instead super would be called with `"Dictionary Words"` provided.

File: README.md (hunk 1 above)
Comment (Piccirello) [reply README.md]: Updated to snake_case

File: src/filters/dictionary.js (hunk 3 above)
Comment (Piccirello) [reply src/filters/dictionary.js]: Moved the name into the constructor, and the derived classes no longer take any arguments

**Pull request merged**